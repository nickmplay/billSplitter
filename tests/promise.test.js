let testDiv;

const proUpdate = (dom, fail) => {
  const newInner = "something else";
  return new Promise((resolve, reject)=>{
    if(!fail){
      dom.innerHTML = newInner;
    }
    if( dom.innerHTML == newInner){
      resolve(dom);
    } else {
      reject("dom not updated");
    }
  });  
}; 

beforeEach(()=>{
  testDiv = document.createElement("div");
  testDiv.innerHTML = "Say something";
});

test('see div snapshot', () => {
  expect(testDiv).toMatchSnapshot();
});

test('promise should update', (done)=>{
  proUpdate(testDiv, false).then(
    (newDom) => {
      expect(newDom).toMatchSnapshot();
      done();
    }
  ).catch(
    (e)=>{
      expect(e).toBe("dom not updated");
      expect(testDiv).toMatchSnapshot();
      done();
    }
  );
});

test('promise should not update', (done)=>{
  proUpdate(testDiv, true).then(
    (newDom) => {
      expect(newDom).toMatchSnapshot();
      done();
    }
  ).catch(
    (e)=>{
      expect(e).toBe("dom not updated");
      expect(testDiv).toMatchSnapshot();
      done();
    }
  );
}); 