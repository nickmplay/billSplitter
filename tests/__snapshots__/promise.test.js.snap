// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`promise should not update 1`] = `
<div>
  Say something
</div>
`;

exports[`promise should update 1`] = `
<div>
  something else
</div>
`;

exports[`see div snapshot 1`] = `
<div>
  Say something
</div>
`;
